### ISSUE TITLE
Remove timestamp from feedback google sheet data?

### ISSUE THREAD

**Issue by knod** [codeforboston/cliff-effects#751]
I could see how that could be used to identify a person, especially when the app is new.

**Comment by ethanbb** [comment]
Hm, maybe. I would argue that it still wouldn't be that hard to identify someone if we're only testing with a few users at a time and checking for new feedback after each session.

If we do decide to do this though, the way we have it set up now makes it a little hard... From https://developers.google.com/apps-script/guides/triggers/installable#google_apps_triggers

>Script executions and API requests do not cause triggers to run. For example, calling FormResponse.submit() to submit a new form response does not cause the form's submit trigger to run.

We could maybe instead use an "open" trigger for the spreadsheet (although that only runs if the person opening it has permission to edit), or a time-based trigger, to clear the "Timestamp" column. I don't think there's a way to prevent form submissions from being timestamped in the first place, but I'm not sure.

**Comment by dylanesque** [comment]
I'll be researching this over the next few days, the fix may be as simple as tweaking the formula to just show the date.

**Comment by dylanesque** [comment]
So, it turns out that you can hide a column pretty easily in Google Sheets. If all users that we *don't* want to see timestamps don't have write access, this should be pretty airtight,

**Comment by knod** [comment]
Hmm, that's a simple solution visually, but can the data for that still be found?

**Comment by dylanesque** [comment]
Yes, but only by users with edit privileges only. Hard to say for certain since the official docs seem.....non-existent?

**Comment by knod** [comment]
Can the page be scraped for that information?

**Comment by dylanesque** [comment]
Possibly. Does this information end up on the sheet automatically? I feel like if the information creates that much of a liability, it shouldn't be there period.

**Comment by knod** [comment]
It does end up there automatically. Google does it.

**Comment by ethanbb** [comment]
Honestly, perfect security isn't possible. I looked at the console on Google Sheets for a few minutes and as far as I can tell, it's pretty opaque - might be possible to extract data through there but it's not clear how. Maybe there's an easier way to see data that's been hidden without editing. But at some point I feel we have to ask what type of "attacker" we're really protecting against. 

The biggest threat is probably ourselves, when reviewing feedback, associating comments with the people who made them based on the time or order of submission, even if we don't intend to. Hiding the timestamps helps with that (although we could still accidentally unhide them, so that's not ideal). If we could also randomize the order each time somehow, that would obscure the order of submission - not sure how easy that would be for a sheet managed by Google Forms though.

**Comment by knod** [comment]
Even with the column hidden, that is.

**Comment by ethanbb** [comment]
Hm, I see one timestamp as part of what looks like a description of the sheet for previews or something, but it cuts off after just a little bit of the first client data structure. This was by doing a Ctrl-F for "2018". Is that what you found or did I miss something?

**Comment by knod** [comment]
No, not that. It's in places like this:
```js
[
  null, null, 3,
  [ null, 3, null, 43213.94681386574 ],
  null, null, 2
]
```
That number is a legacy timestamp created from a bug in Lotus way, way back. It's the number of days since Dec 30, 1899. https://stackoverflow.com/a/3963650

**Comment by knod** [comment]
If you want, you can put some nonsense word in a column, hide it, and then search for that.

**Comment by ethanbb** [comment]
Whoa OK, that's pretty wild! I guess it makes sense that it would store dates relative to some epoch rather than in human-readable form though.

**Comment by knod** [comment]
This actually seems like it might be an artifact of using a form as a middleman between our code and the google sheets. Might be worth trying this instead: https://medium.com/@dmccoy/how-to-submit-an-html-form-to-google-sheets-without-google-forms-b833952cc175
